dvc/repo/plots/__init__.py

```diff
@@ -317,11 +317,16 @@ def _deduplicate(lst: List):
         return list({elem: None for elem in lst}.keys())
 
     y = config.get("y", None)
+
     if isinstance(y, dict):
         sources = list(y.keys())
     else:
         sources = [plot_id]
 
+    x = config.get("x", None)
+    if isinstance(x, dict):
+        sources.append(first(x.keys()))
+
     return _deduplicate(source for source in sources)
```

Comment:
`_deduplicate` could be replaced by `from funcy import distinct` https://funcy.readthedocs.io/en/stable/seqs.html#distinct

Reply:
Done